boofuzz/socket_connection.py

```diff
@@ -165,13 +167,23 @@ def open(self):
 
         # if SSL is requested, then enable it.
         if self.proto == "ssl":
-            if self.server:
-                ssl_sock = ssl.wrap_socket(self._sock, keyfile=self.keyfile, certfile=self.certfile,
-                                                      server_side=True)
-                self._sock = ssl_sock
-            else:
-                ssl_sock = ssl.wrap_socket(self._sock)
-                self._sock = ssl_sock
+            if not self.sslcontext:
+                # If we are the SSL client and user did not give us a SSLContext,
+                # then we just use a default one. However, we only can verify the
+                # identity of the SSL server, if we have been given a server_hostname.
+                if not self.server and not self.sslcontext:
+                    self.sslcontext = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+                    if self.server_hostname:
+                        self.sslcontext.check_hostname = True
+                        self.sslcontext.verify_mode = ssl.CERT_REQUIRED
+                    else:
+                        # Only option left would be to accept any certificates.
+                        # This would left us vulnerable to man-in-the-middle attacks.
+                        # We won't allow that and the user has to remove the assert below.
+                        assert(False)
```

Comment:
Let's avoid requiring code modification just to use the framework. Sounds like the user already needs to make their own context anyway.

Reply:
Yes, the user will need to make his own context, if he wants to accept any certificates. This is also explained in the ```fuzz-ssl-server.py``` example.